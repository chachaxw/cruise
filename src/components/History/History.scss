@import '../../styles/_variables';

.cruise-history {
    max-width: 100%;
    padding: 10px;
    box-sizing: border-box;
    background-color: $menuBackground;

    h1 {
        font-size: 24px;
        color: #cccccc;
    }

    .history-list {
        color: #999999;
        font-size: 12px;
        margin-top: 15px;
        max-height: 240px;
        overflow-y: auto;
        list-style: outside;

        .history-item {
            display: flex;
            align-items: center;
            margin-bottom: 10px;

            &:hover {
                color: $cyanBlue;
            }
        }

        .item-dot {
            display: inline-block;
            width: 4px;
            height: 4px;
            border-radius: 50%;
            margin-right: 14px;
            background-color: #999999;
        }

        p {
            max-width: 200px;
            text-overflow: ellipsis;
            white-space: nowrap;
            overflow: hidden;
        }
    }
}