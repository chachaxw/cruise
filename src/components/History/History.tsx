import * as React from 'react';
import './History.scss';

export type HistoryItem = {
    id: number,
    name: string,
    status: string,
}

interface InternalProps {
    history: HistoryItem[],
}

export default class History extends React.Component<InternalProps> {

    public renderHistoryItem(props: HistoryItem) {
        return (
            <li className="history-item" key={props.id}>
                <span className="item-dot"></span>
                <p>{props.name}/{props.status}</p>
            </li>
        );
    }

    public render() {
        const { history } = this.props;

        return (
            <div className="cruise-history">
                <h1>History</h1>
                <ul className="history-list">
                    {history && history.length && history.map((item, index) => this.renderHistoryItem(item))}
                </ul>
            </div>
        );
    }
}
