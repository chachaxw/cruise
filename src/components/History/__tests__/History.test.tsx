import * as React from 'react';
import { StaticRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import History from '../History';

it('renders correctly with given prop history', () => {
    const test = [
        { id: 0, name: 'Test Agent1', status: 'Test Status1' },
        { id: 1, name: 'Test Agent2', status: 'Test Status2' },
        { id: 2, name: 'Test Agent3', status: 'Test Status3' },
    ];

    const testRenderer = TestRenderer.create(
        <StaticRouter context={{}}>
            <History history={test}></History>
        </StaticRouter>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
});