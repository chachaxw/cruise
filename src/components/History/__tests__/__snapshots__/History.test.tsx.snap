// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with given prop history 1`] = `
<div
  className="cruise-history"
>
  <h1>
    History
  </h1>
  <ul
    className="history-list"
  >
    <li
      className="history-item"
    >
      <span
        className="item-dot"
      />
      <p>
        Test Agent1
        /
        Test Status1
      </p>
    </li>
    <li
      className="history-item"
    >
      <span
        className="item-dot"
      />
      <p>
        Test Agent2
        /
        Test Status2
      </p>
    </li>
    <li
      className="history-item"
    >
      <span
        className="item-dot"
      />
      <p>
        Test Agent3
        /
        Test Status3
      </p>
    </li>
  </ul>
</div>
`;
